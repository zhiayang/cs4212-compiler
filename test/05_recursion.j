// 05_recursion.j

class Main
{
	Void main()
	{
		println(new Foo().sums(100));
	}
}

class Foo
{
	Int sums(Int num)
	{
		return sums(0, num);
	}

	Int sums(Int acc, Int num)
	{
		if(num == 0) { return acc; }
		else         { println(acc); return sums(acc + num, num - 1); }
	}
}
