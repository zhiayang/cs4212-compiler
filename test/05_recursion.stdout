0
100
199
297
394
490
585
679
772
864
955
1045
1134
1222
1309
1395
1480
1564
1647
1729
1810
1890
1969
2047
2124
2200
2275
2349
2422
2494
2565
2635
2704
2772
2839
2905
2970
3034
3097
3159
3220
3280
3339
3397
3454
3510
3565
3619
3672
3724
3775
3825
3874
3922
3969
4015
4060
4104
4147
4189
4230
4270
4309
4347
4384
4420
4455
4489
4522
4554
4585
4615
4644
4672
4699
4725
4750
4774
4797
4819
4840
4860
4879
4897
4914
4930
4945
4959
4972
4984
4995
5005
5014
5022
5029
5035
5040
5044
5047
5049
5050
